docs/src/main/tut/freemonad.md

```diff
@@ -77,30 +77,21 @@ recursive values.
 We need to create an ADT to represent our key-value operations:
 
 ```tut:silent
-sealed trait KVStoreA[+Next]
-case class Put[T, Next](key: String, value: T, next: Next) extends KVStoreA[Next]
-case class Get[T, Next](key: String, onResult: T => Next) extends KVStoreA[Next]
-case class Delete[Next](key: String, next: Next) extends KVStoreA[Next]
+sealed trait KVStoreA[A]
+case class Put[T](key: String, value: T) extends KVStoreA[Unit]
+case class Get[T](key: String) extends KVStoreA[Option[T]]
+case class Delete(key: String) extends KVStoreA[Unit]
 ```
 
-The `next` field in each of the types provides a way to link an
-operation with successive values. The `Next` type parameter can be
-anything at all, including `Unit`. It can be thought of as a carrier,
-a way to link a single operation with successive operations.
-
-As we will see, the `next` field is also necessary to allow us to
-provide a `Functor` instance for `KVStoreA[_]`.
-
 ### Free your ADT
 
 There are six basic steps to "freeing" the ADT:
 
 1. Create a type based on `Free[_]` and `KVStoreA[_]`.
-2. Prove `KVStoreA[_]` has a functor.
-3. Create smart constructors for `KVStore[_]` using `liftF`.
-4. Build a program out of key-value DSL operations.
-5. Build a compiler for programs of DSL operations.
-6. Execute our compiled program.
+2. Create smart constructors for `KVStore[_]` using `liftF`.
```

Comment:
An alternative to `liftF` is to use `Free.inject` when targeting `cats.data.Coproduct` as a composition mechanism maybe it should be mentioned here.